shared/src/api/index.coffee

```diff
@@ -165,13 +165,16 @@ class APIHandlerBase
 
     onSuccess ?= DEFAULT_SUCCESS
     onFail ?= DEFAULT_FAIL
-    return new Promise (resolve) =>
+    return new Promise (resolve, reject) =>
+
       delay =>
         @_xhr.request(requestConfig)
+          .then(((resp) -> resolve(resp); resp), ((resp) -> reject(resp); Promise.reject(resp)))
```

Comment:
this is a pretty drastic change, does it still work if moved back as the last `then` with the `reject` added?  also, did you try using the `onFail` api setting ?